packages/core/fs/src/MemoryFS.js

```diff
@@ -529,18 +535,23 @@ export class MemoryFS implements FileSystem {
   }
 
   _registerWorker(handle: Handle) {
-    this.workerHandles.push(handle);
+    this._workerHandles.push(handle);
+    if (this._numWorkerInstances === this._workerHandles.length) {
+      this._emitter.emit('allWorkersRegistered');
+    }
   }
 
   async _sendWorkerEvent(event: WorkerEvent) {
     // Wait for worker instances to register their handles
-    while (this.workerHandles.length < this.numWorkerInstances) {
-      await sleep(0);
+    while (this._workerHandles.length < this._numWorkerInstances) {
+      await new Promise(resolve => this._workerRegisterResolves.push(resolve));
```

Comment:
You might be able to do something like this?

```js
await new Promise(resolve => {
  this._emitter.on('allWorkersRegistered', resolve)
});
```

The would avoid the extra array of promises.

Reply:
I was worried about adding a listener for every event. I tried having the listener remove itself once it no longer need to listen, but that didn't seem to work. Sorry, I merged before I saw your comment.